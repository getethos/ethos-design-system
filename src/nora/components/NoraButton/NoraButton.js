import React from 'react'
import PropTypes from 'prop-types'
import styles from './NoraButton.module.scss'

export const NoraButton = ({ className, children }) => {
  return (
    <button className={[styles.Button, className].join(' ')}>{children}</button>
  )
}

NoraButton.propTypes = {
  className: PropTypes.string,
  children: PropTypes.string.isRequired,
}
