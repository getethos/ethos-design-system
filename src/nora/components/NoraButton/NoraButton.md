Default NoraButton
```jsx
<NoraButton>Submit</NoraButton>
```

NoraButton with custom stying passed in
```jsx
import styles from './NoraButtonExample.module.scss';
<NoraButton className={styles.DemoWidth}>Submit</NoraButton>
```

