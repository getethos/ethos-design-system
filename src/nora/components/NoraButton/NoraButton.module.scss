.Button {
  font-weight: 500;
  font-size: 13px;
  height: var(--Space-32);
  line-height: var(--Space-32);
  letter-spacing: 0.015em;
  color: white;
  background: var(--NoraBlue-400);
  border-radius: 3px;
  text-align: center;
  display: inline-block;
}

/* Used only for md example */
.DemoWidth {
  width: var(--Space-80);
}