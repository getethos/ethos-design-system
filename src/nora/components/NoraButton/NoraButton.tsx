import * as React from 'react'
import styles from './NoraButton.module.scss'
type NoraButtonProps = {
  'data-tid'?: string
  disabled?: boolean
  type?: 'button' | 'submit'
  name?: string
  onClick?: (...args: any[]) => any
  role?: string
  className?: string
}
export const NoraButton: React.FC<NoraButtonProps> = ({
  className,
  disabled,
  type,
  name,
  onClick,
  role,
  children,
  ...rest
}) => {
  const cssClasses = [styles.Button, className].join(' ')
  return (
    <button
      className={cssClasses}
      disabled={disabled}
      type={type}
      name={name}
      onClick={onClick}
      data-tid={rest['data-tid']}
      role={role}
    >
      {children}
    </button>
  )
}
