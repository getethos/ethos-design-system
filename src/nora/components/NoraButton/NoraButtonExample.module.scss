.DemoWidth {
  width: var(--Space-80);
}