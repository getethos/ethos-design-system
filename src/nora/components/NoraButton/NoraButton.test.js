import React from 'react'
import renderer from 'react-test-renderer'
import { NoraButton } from './NoraButton'
import styles from './NoraButton.module.scss'

describe('NoraButton', () => {
  it('renders default NoraButton', () => {
    const tree = renderer.create(<NoraButton>default</NoraButton>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('allows className to be passed as a prop', () => {
    const tree = renderer
      .create(<NoraButton className={styles.DemoWidth}>customized</NoraButton>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
