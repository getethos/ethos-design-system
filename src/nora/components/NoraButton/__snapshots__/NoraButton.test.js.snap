// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoraButton allows className to be passed as a prop 1`] = `
<button
  className="Button DemoWidth"
>
  customized
</button>
`;

exports[`NoraButton renders default NoraButton 1`] = `
<button
  className="Button "
>
  default
</button>
`;
